Dockerfile

```diff
@@ -23,8 +23,17 @@ RUN \
     command -v lvm && \
     command -v mdadm
 
+# Install the Amazon SSM Agent
+RUN \
+    dnf -yq install 'dnf-command(download)' cpio rsync && \
+    dnf download amazon-ssm-agent && \
```

Comment:
This works, but its a bit roundabout. Have you considered re-using the code from the [control container](https://github.com/bottlerocket-os/bottlerocket-control-container/blob/develop/Dockerfile#L81)? The rpm it pulls down doesn't seem to require as many dependencies as the one in the dnf repos.

Reply:
Yes I in fact did but I don't think I saw a real benefit to using that approach. As a matter of fact, when Ben asked me to move the ssm-agent download from the script to the Dockerfile, the first thing that came to mind was to see how the control container does it, so I ran the image build after importing the SSM agent install code from that Dockerfile (the same one you linked). That image turned out to be about 150 MB bigger than the one from DNF.

To further address the image size issue, I have added an intermediate step in the Dockerfile to install SSM agent and only copying that to the final stage. I also reduced the number of `RUN` and `COPY` instructions in the image, bringing the image size to about 130MB.

Reply:
This is great! I should look into doing this for the control container as well.